File: backend/src/main/java/com/project2/domain/post/mapper/CommentMapper.java
package com.project2.domain.post.mapper;

import org.springframework.stereotype.Component;

import com.project2.domain.member.entity.Member;
import com.project2.domain.post.dto.comment.CommentRequestDTO;
import com.project2.domain.post.dto.comment.CommentResponseDTO;
import com.project2.domain.post.entity.Comment;
import com.project2.domain.post.entity.Post;

@Component
public class CommentMapper {

	public Comment toEntity(CommentRequestDTO request, Post post, Member member, Comment parent) {
		int depth = (parent == null) ? 0 : 1;
		return Comment.builder()
			.content(request.getContent())
			.depth(depth)
			.post(post)
			.member(member)
			.parent(parent)
			.build();
	}

	public CommentResponseDTO toResponseDTO(Comment comment, String nickname) {
		return new CommentResponseDTO(
			comment.getId(),
			comment.getContent(),
			nickname,
			comment.getParent() != null ? comment.getParent().getId() : null
		);
	}

}

File: backend/src/main/kotlin/com/project2/domain/post/mapper/CommentMapper.kt
package com.project2.domain.post.mapper

import com.project2.domain.member.entity.Member
import com.project2.domain.post.dto.comment.CommentRequestDTO
import com.project2.domain.post.dto.comment.CommentResponseDTO
import com.project2.domain.post.entity.Comment
import com.project2.domain.post.entity.Post
import org.springframework.stereotype.Component

@Component
class CommentMapper {

    fun toEntity(request: CommentRequestDTO, post: Post, member: Member, parent: Comment?): Comment {
        val depth = if (parent == null) 0 else 1
        return Comment().apply {
            content = request.content
            this.post = post
            this.member = member
            this.parent = parent
            this.depth = depth
        }
    }

    fun toResponseDTO(comment: Comment, nickname: String): CommentResponseDTO {
        return CommentResponseDTO(
            id = comment.id,
            content = comment.content,
            nickname = nickname,
            parentId = comment.parent?.id
        )
    }
}
